cmd/shortener/services/shorter.go

```diff
@@ -14,24 +14,53 @@ type Shorter interface {
 	RetrieveURL(id string) (string, error)
 	RetrieveURLsForUser(id string) ([]entities.LinkItem, error)
 	Ping() error
+	DeleteBatch(keys []string, userID string) error
 }
 
 var (
 	ErrEmptyURL     = errors.New("url is empty")
 	ErrEmptyID      = errors.New("id is empty")
 	ErrIDIsNotFound = errors.New("id is not found")
+	ErrDeletedID    = errors.New("id deleted")
 )
 
 type ShorterURL struct {
-	repository repositories.Repository
+	repository    repositories.Repository
+	deleteChannel chan entities.DeleteCandidate
+}
+
+func (s *ShorterURL) DeleteBatch(keys []string, userID string) error {
+	urls := s.repository.GetAllFor(userID)
+	if len(urls) == 0 {
+		return errors.New("No such userID: " + userID)
+	}
+
+	for _, item := range urls {
+		log.Printf("%v =>", item)
+		for _, id := range keys {
+			log.Printf("<= %v", id)
+			if id == item.ShortURL {
+				go func(id string) {
+					log.Printf("send for deletion id: %v \n", id)
+					s.deleteChannel <- entities.DeleteCandidate{Key: id, UserID: userID}
+				}(id)
+
+			}
+		}
+	}
```

Comment:
чтобы получился реально batch delete можно было бы сделать примерно так :
```suggestion
s.deleteChannel <- entities.DeleteCandidate{Keys: keys, UserID: userID}
```